### PR TITLE
bugfix: Update Bookmarks with Singer Utils Package

### PR DESCRIPTION
**Pull request by pnadolny13**
Someone in the [Meltano community slack](https://meltano.slack.com/archives/C01TCRBBJD7/p1664207537554719) raised a bug with this tap related to a key error when accessing the bookmarks. I noticed that the state dict/bookmarks were being edited directly which was causing the issue and theres a nice singer utility to get/set bookmark values already. So I implemented that in exchange. I think if you always provide state with bookmarks present then this wouldnt throw an error but on the first run with no state it looks like it does. Although I dont have maestro qa api keys to test it myself, these changes seemed to resolve the problem for the community member.

Also start_date wasnt a required config key but if you dont have state then this would fail is start_date is missing. Personally I'd consider that a requirement then.

@kyuhanpathlight @k3llymariee What do you think? (I saw you were the last commiters)

### CODE REVIEW

**Review by kyuhanpathlight — APPROVED** [review]
Excellent.  This looks good to me.  Thanks for your contribution!

**Comment by pnadolny13** [comment]
@kyuhanpathlight great thanks! I dont have merge permission so you can feel free to hit the merge button whenever you're comfortable with the changes 😄 .

**Comment by zachwilner** [comment]
thanks @pnadolny13 and @kyuhanpathlight  for doing this!

**Comment by pnadolny13** [comment]
@kyuhanpathlight would you be able to merge this in for me when you get a chance? Thanks

**Comment by zachwilner** [comment]
@kyuhanpathlight any chance we can merge this?

**Pull request merged**
